flink-tests/src/test/java/org/apache/flink/test/accumulators/AccumulatorLiveITCase.java

```diff
@@ -109,125 +123,157 @@ public void testProgram() throws Exception {
 			ExecutionEnvironment env = new PlanExtractor();
 			DataSet<String> input = env.fromCollection(inputData);
 			input
-					.flatMap(new Tokenizer())
 					.flatMap(new WaitingUDF())
 					.output(new WaitingOutputFormat());
 			env.execute();
 
-			/** Extract job graph **/
+			// Extract job graph and set job id for the task to notify of accumulator changes.
 			JobGraph jobGraph = getOptimizedPlan(((PlanExtractor) env).plan);
+			jobID = jobGraph.getJobID();
+
+			// register for accumulator changes
+			jobManager.tell(new TestingJobManagerMessages.NotifyWhenAccumulatorChange(jobID), getRef());
+			expectMsgEquals(true);
```

Comment:
When you wait for messages then this should always happen within a `within` block. Otherwise this might never terminate in a failure case.